### PR TITLE
Refactor/prepare async class level scanner

### PR DESCRIPTION
**Pull request by timonback**

### CODE REVIEW

**Review by sam0r040 — COMMENTED** [review]
I really like the new structure and naming, it should make things more clear now. 
I left some suggestion to make some things more clear through naming. 
Overall I'm not sure about creating dependencies in the constructor, it makes class interface bigger than they need to be and hides the real dependency (not sure if this is intended or not).

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/channels/ChannelsInClassScanner.java [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+package io.github.springwolf.core.asyncapi.scanners.channels;
 
 import io.github.springwolf.asyncapi.v3.model.channel.ChannelObject;
 
 import java.util.Map;
 import java.util.stream.Stream;
 
-public interface SpringAnnotationChannelsScannerDelegator {
+public interface ChannelsInClassScanner {
     Stream<Map.Entry<String, ChannelObject>> scan(Class<?> clazz);
```
[the remark this answers is not in the document]
Comment (timonback) [reply springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/channels/ChannelsInClassScanner.java]: Good idea, I changed in to List.
When we have an operationId, we can do the same thing for operations as well

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/annotation/AnnotationScannerUtil.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+public class AnnotationScannerUtil {
+
+    private AnnotationScannerUtil() {}
+
+    /**
+     * Find all annotated methods on an annotated class
+     *
+     * Transform is only called if methods are found
+     */
+    public static <C extends Annotation, M extends Annotation, R> Stream<R> findAnnotatedMethods(
```
[the remark this answers is not in the document]
Comment (timonback) [reply springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/annotation/AnnotationScannerUtil.java:28]: Any suggestions?
I thought about using the same name for both methods, as the function gets all methods and the different parameters (class + method annotation) kinda act as filters

File: springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/annotation/AnnotationScannerUtil.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            Class<?> clazz,
+            Class<C> classAnnotationClass,
+            Class<M> methodAnnotationClass,
+            BiFunction<Class<?>, Set<MethodAndAnnotation<M>>, Stream<R>> transformer) {
+        log.debug("Scanning class \"{}\" for @\"{}\" annotation", clazz.getName(), classAnnotationClass.getName());
+        Set<MethodAndAnnotation<M>> methods = Stream.of(clazz)
```
[the remark this answers is not in the document]
Comment (timonback) [reply springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/annotation/AnnotationScannerUtil.java:34]: No other reason, I just like the streaming API.
If this Stream.of(one) looks artificial, I open open for alternatives

**Comment by timonback** [comment]
> I really like the new structure and naming, it should make things more clear now. I left some suggestion to make some things more clear through naming. Overall I'm not sure about creating dependencies in the constructor, it makes class interface bigger than they need to be and hides the real dependency (not sure if this is intended or not).

Just updated the PR.

I agree on the dependencies in constructors and thereby throwing away the advantage of dependency injection. Took some time to refactor further, but is included now.

**Pull request merged**